server/src/main/java/org/elasticsearch/common/util/BigArrays.java

```diff
@@ -58,15 +58,9 @@ public static long overSize(long minTargetSize, int pageSize, int bytesPerElemen
 
         long newSize;
         if (minTargetSize < pageSize) {
-            newSize = ArrayUtil.oversize((int)minTargetSize, bytesPerElement);
+            newSize = Math.min(ArrayUtil.oversize((int) minTargetSize, bytesPerElement), pageSize);
```

Comment:
This now doesn't always return a multiple of `pageSize`; is that deliberate? Maybe it's ok, but if it is ok then why do we put effort into rounding `newSize` up to the next multiple of `pageSize` in the `else` branch?

Reply:
> This now doesn't always return a multiple of pageSize; is that deliberate? 

This was the case before as well. For small allocations (those for which `Arrays.oversize` does return less than a page) we return the result of oversize.

> Maybe it's ok, but if it is ok then why do we put effort into rounding newSize up to the next multiple of pageSize in the else branch?

I think the reasoning is that we only want to use a full page for allocations above a certain size. But once we get to large allocations we only want to use full pages.
For small pages I mostly retained the behavior here. Except for the case of say 15k with 16k pages, where previously `oversize` would spit out something like 17k and we'd then go for 2 pages, whereas now we go for a single page.
But as long as oversize stays below a full page in size the behavior should not change with this PR.

Reply:
Ah ok I misinterpreted `min` as `max`. I see now, thanks.